pkg/core/tekton/builder/task_spec.go

```diff
@@ -0,0 +1,71 @@
+package builder
+
+import (
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// TaskSpecBuilder holds the tekton task spec definition.
+type TaskSpecBuilder struct {
+	TaskSpec v1beta1.TaskSpec
+}
+
+// NewTaskSpecBuilder creates a TaskSpec with default values.
+func NewTaskSpecBuilder(name, image, command string) *TaskSpecBuilder {
+	b := &TaskSpecBuilder{}
+	b.TaskSpec = v1beta1.TaskSpec{
+		Steps: []v1beta1.Step{{Container: corev1.Container{
+			Name:    name,
+			Image:   image,
```

Comment:
What's the difference between this image and the one set by `TaskImage` ?

Reply:
`TaskImage` adds the option to overwrite the step image. This is used, for example, when the image comes from an output of a previous task, which in that case the Task needs to receive it as a parameter so the image is set to `$(params.IMAGE)`

Reply:
got it, thanks :)